src/gadgets/r1cs.rs

```diff
@@ -229,7 +229,7 @@ impl<E: Engine, const N: usize> AllocatedRelaxedR1CSInstance<E, N> {
     n_limbs: usize,
   ) -> Result<Self, SynthesisError> {
     // Compute r:
-    let mut ro = E::ROCircuit::new(ro_consts, NUM_FE_FOR_RO);
+    let mut ro = E::ROCircuit::new(ro_consts, 7 + N);
```

Comment:
Please introduce another constant and avoid the magic number.

Reply:
I noticed in trying to fix this that our use of `NUM_FE_WITHOUT_IO_FOR_CRHF` has drifted away from its original purpose, so I'm going to try to fix up the use of constants beyond just adding this one.